epan/dissectors/packet-ippusb.c

```diff
@@ -25,66 +25,104 @@
 #include "packet-usb.h"
 #include <stdio.h>
 
+#define HTTP 0
+#define IPP 1
+
 void proto_register_ippusb(void);
 void proto_reg_handoff_ippusb(void);
+static int is_http_header(int first_linelen, const guchar *first_line); 
 
 static int proto_ippusb = -1;
+static gint ett_ippusb = -1;
+static gint ett_ippusb_as = -1;
+static gint ett_ippusb_attr = -1;
+static gint ett_ippusb_member = -1;
+
 static dissector_table_t ippusb_dissector_table;
 
 static int
 dissect_ippusb(tvbuff_t *tvb, packet_info *pinfo, proto_tree *tree, void* data)
 {
     //skip to http header info
     //put dummy info in for http dissector
-
-    proto_tree  *ippusb_tree = tree;
-    //proto_item  *ti;
-    int         offset     = 0;
-    //http_message_info_t *message_info = (http_message_info_t *)data;
-    //gboolean    is_request;
-    //guint16     operation_status;
-   // const gchar *status_type;
-    //guint32	request_id;
-    //conversation_t *conversation;
-
-    //offset +=
-
-    if (tvb_captured_length_remaining(tvb, offset) > 0) {
-        next_tvb = tvb_new_subset_remaining(tvb, offset);
-        offset += try_dissect_next_protocol(parent, next_tvb, pinfo, usb_conv_info, urb_type, tree, NULL);
+    proto_tree  *ippusb_tree;
+    proto_item  *ti;
+    int offset = 0;
+    int ret;
+    int first_linelen;
+    const guchar *first_line;
+    gint next_offset;
+
+    ti = proto_tree_add_item(tree, proto_ippusb, tvb, offset, -1, 0);
+    ippusb_tree = proto_item_add_subtree(ti, ett_ippusb);
+
+    first_linelen = tvb_find_line_end(tvb, offset,  tvb_ensure_captured_length_remaining(tvb, offset), &next_offset, TRUE);
+    first_line = tvb_get_ptr(tvb, offset, first_linelen);
+
+    //check if there is a http header to parse before ipp
+    if (is_http_header(first_linelen, first_line)) {        
+        ret = dissector_try_uint_new(ippusb_dissector_table, HTTP, tvb, pinfo, tree, TRUE, data);
+    }
+    else {
+        ret = dissector_try_uint_new(ippusb_dissector_table, IPP, tvb, pinfo, tree, TRUE, data);
     }
 
-    if (tvb_captured_length_remaining(tvb, offset) > 0) {
-        /* There is still leftover capture data to add (padding?) */
-        proto_tree_add_item(parent, hf_usb_capdata, tvb, offset, -1, ENC_NA);
+    if (ret) {
+            return tvb_captured_length(tvb);
     }
+    else {
+        return 0;
+    }
+}
 
-    if (tvb_offset_exists(tvb, offset) && data) {
-        call_data_dissector(tvb_new_subset_remaining(tvb, offset), pinfo, ippusb_tree);
+static int 
+is_http_header(int first_linelen, const guchar *first_line) {
+    if ((first_linelen >= 5 && strncmp(first_line, "HTTP/", 5) == 0) ||
+		(first_linelen >= 3 && strncmp(first_line, "ICY", 3) == 0) ||
```

Comment:
For what I can see you used one less tab to separate sections by line length. This one seems off though given that format.